server/controllers/users.js

```diff
@@ -299,10 +299,28 @@ module.exports = function(app) {
    */
   return {
 
+    getOrCreate: (email, cb) => {
```

Comment:
This looks like a security hole, you allow to get and modify an existing user just by providing an email address? Shouldn't the email address instead be taken from the authenticated user?